pyquil/_parser/Quil.g4

```diff
@@ -169,7 +169,7 @@ POWER               : '^' ;
 
 // Identifiers
 
-IDENTIFIER          : [A-Za-z_] [A-Za-z0-9\-_]* ;
+IDENTIFIER          : [A-Za-z_] ([A-Za-z0-9_] | '-' [A-Za-z]) * ;
```

Comment:
Nitpick: no space before kleene, also can we add a comment describing the rules for identifiers?